core/src/main/scala/org/apache/spark/MapOutputStatistics.scala

```diff
@@ -23,5 +23,9 @@ package org.apache.spark
  * @param shuffleId ID of the shuffle
  * @param bytesByPartitionId approximate number of output bytes for each map output partition
  *   (may be inexact due to use of compressed map statuses)
+ * @param numberOfOutput number of output for each pre-map output partition
```

Comment:
It is different with `bytesByPartitionId`. It is not the number of records for each reducer. Rather said, it is the number of records for each map task.